nautobot_golden_config/forms.py

```diff
@@ -521,47 +506,115 @@ class ConfigPlanCreateForm(BootstrapMixin, forms.Form):
-            "Enter your configuration template here representing CLI configuration.<br>"
-            'You may use Jinja2 templating. Example: <code>{% if "foo" in bar %}foo{% endif %}</code><br>'
-            "You can also reference the device object with <code>obj</code>.<br>"
-            "For example: <code>hostname {{ obj.name }}</code> or <code>ip address {{ obj.primary_ip4.host }}</code>"
-        ),
-        required=True,
-    )
+        model = models.ConfigPlan
+        fields = (
+            "plan_type",
+            "change_control_id",
+            "feature",
+            "commands",
+            "tenant",
+            "location",
+            "region",
+            "site",
+            "rack_group",
+            "rack",
+            "role",
+            "manufacturer",
+            "platform",
+            "device_type",
+            "device",
+            "tag",
+            "status",
+        )
 
 
 class ConfigPlanUpdateForm(NautobotModelForm):
     """Form for ConfigPlan instances."""
 
+    change_control_id = forms.CharField(required=False, label="Change Control ID")
+    change_control_url = forms.URLField(required=False, label="Change Control URL")
+    status = utilities_forms.DynamicModelChoiceField(
+        queryset=Status.objects.all(),
+        query_params={"content_types": models.ConfigPlan._meta.label_lower},
+        required=False,
+    )
+    tag = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=Tag.objects.all(), query_params={"content_types": "dcim.device"}, required=False
+    )
+
     class Meta:
         """Boilerplate form Meta data for ConfigPlan."""
 
         model = models.ConfigPlan
         fields = (
             "change_control_id",
-            "tags",
+            "tag",
             "status",
         )
 
 
+class ConfigPlanFilterForm(NautobotFilterForm):
+    """Filter Form for ConfigPlan."""
+
+    model = models.ConfigPlan
+
+    q = forms.CharField(required=False, label="Search")
+    device_id = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=Device.objects.all(), required=False, null_option="None", label="Device"
+    )
+    created__lte = forms.DateTimeField(label="Created Before", required=False, widget=DatePicker())
+    created__gte = forms.DateTimeField(label="Created After", required=False, widget=DatePicker())
+    plan_type = forms.ChoiceField(
+        choices=add_blank_choice(ConfigPlanTypeChoice), required=False, widget=forms.Select(), label="Plan Type"
+    )
+    feature = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=models.ComplianceFeature.objects.all(),
+        required=False,
+        null_option="None",
+        label="Feature",
+        to_field_name="name",
+    )
+    change_control_id = forms.CharField(required=False, label="Change Control ID")
+    job_result_id = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=JobResult.objects.filter(config_plan__isnull=False).distinct(),
```

Comment:
I have a lot of questions about this, not really sure I understand tbh.

Reply:
Put simply:
`forms.ModelMultipleChoiceField` = a static field that takes in a queryset to populate a choice dropdown. In this case a filter on the actual queryset matters.
`utilities_forms .DynamicModelMultipleChoiceField` = a dynamic field that queries the API for the model provided (and uses query params if applicable).

So, passing the filtered queryset to the dynamic field here doesn't actually do anything since it just uses the model to determine the API call to make.

Reply:
So @joewesch you're saying the Dynamic nature of this choice field we should query all devices and filter using query_params which is in the code right now, more or less just need to remove the orm filter we're calling?